### ISSUE TITLE
Schema is always reset to schema name same as logged username (postgres)

### ISSUE THREAD

**Issue by mikalai-shchurko** [dbeaver/dbeaver#13750]
#### System information: 
- Windows 10 Pro (build 19042.1165)
- DBeaver version: 21.2.0.202108310918 (wasn't reproducible on 6.3.3 and was reproducible at least from 7.0)

#### Connection specification:
- postgres 9.6
- postgresql driver shipped with dbeaver
- Do you use tunnels or proxies (SSH, SOCKS, etc)? - no

#### Describe the problem you're observing:
If username matches any of schema names, selected schema in dropdown is ignored and reset to 'username schema' before query execution.

#### Steps to reproduce, if exist:
1. create postgres server: 
`docker run --name db_Devpp -e POSTGRES_USER=dbowner -e POSTGRES_PASSWORD=dbowner -p 5443:5432 postgres:9.6`
2. create additoinal schema with the same name as username. create the same table in both schemas, fill it with distinctive value:
```
CREATE TABLE public.newtable (
	value varchar NULL
);
insert into public.newtable values ('public');
CREATE SCHEMA dbowner AUTHORIZATION dbowner;
CREATE TABLE dbowner.newtable (
	value varchar NULL
);
insert into dbowner.newtable values ('dbowner');
```
3. refresh structure in a tree and select public schema as default for queries:
![image](https://user-images.githubusercontent.com/9625621/131668812-c25c4f3a-a8dd-48ae-a9e6-5c7cacc47e76.png)
4. execute: `select * from newtable n ;`

Expected: value from public.newtable
Actual: value from dbowner.newtable. And schema selector switched to dbowner schema.
![image](https://user-images.githubusercontent.com/9625621/131669364-ce9e1dac-a9fb-4763-8eb5-7ee0d9b48830.png)

#### Include any warning/errors/backtraces from the logs
No errors in log view.
No related errors in the log files.

**Comment by LonwoLonwo** [comment]
Thanks for the bug report!

**Comment by LonwoLonwo** [comment]
Oh. I see.
PG [documentation](https://www.postgresql.org/docs/current/functions-info.html) says about cureent_schema() method : Returns the name of the schema that is first in the search path.

And we use this expression after each execution in SQL Editor.
We put "$user" in the first place in the search_path, and in case, when the user name is the same as the schema name - We change the scheme to the one that was first in the search_path.

You can change this behaviour in Preferences -> Editors -> SQL Editor -> SQL Processing.
![2021-09-01 17_31_50-Preferences](https://user-images.githubusercontent.com/45152336/131692467-21c6c88a-5943-4fc8-8fbf-28eedf314ab4.png)

**Comment by mikalai-shchurko** [comment]
Thanks @LonwoLonwo
When I uncheck this option it still queries the wrong schema (dbowner instead of public), but doesn't change selected value in dropdown afterwards :) 
So it doesn't solve the main problem.
Here's what I have after execution:
![image](https://user-images.githubusercontent.com/9625621/131701157-d7daabbc-0558-4b24-96ce-37675f186713.png)

**Comment by LonwoLonwo** [comment]
Yeah. It's again because the user is first in your search_path. 
As a workaround for now - you can set your search_path manually.

**Comment by LonwoLonwo** [comment]
Fixed in DBeaver 21.2.4

![138327388-2ae65c55-b795-438a-8916-3f7b587f6745](https://user-images.githubusercontent.com/45152336/138421906-9d8a8052-3866-4f7c-8fab-9f7b83dfd72d.png)

Now we set the $user parameter at the end of the search path.

**Comment by uslss** [comment]
verified
